### PR TITLE
fix: cell out of bounds

### PR DESCRIPTION
**Pull request by keita-determined**
## Description
When the size of experiment list is changed, `data[row]` can be `undefined` due to the size is changed.

https://user-images.githubusercontent.com/109113616/236358180-33f811f6-667c-4e54-821f-b40a647c8f73.mov

## Test Plan
- Check if table doesnt break even if the size of experiment list is changed

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx [hunk 1]
```diff
@@ -259,18 +259,23 @@ export const GlideTable: React.FC<GlideTableProps> = ({
...
+        copyData: '',
+        data: { kind: 'spinner-cell' },
+        kind: GridCellKind.Custom,
+      };
+
+      if (row >= data.length) {
+        return customCell;
+      }
+
       return Loadable.match(data[row], {
```
Comment (keita-determined) [inline webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx:273]: this `data[row]` can be `undefined` when the size of exp list changes

File: webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx [hunk 2]
```diff
@@ -259,18 +259,23 @@ export const GlideTable: React.FC<GlideTableProps> = ({
...
+      const customCell: GridCell = {
+        allowOverlay: true,
+        copyData: '',
+        data: { kind: 'spinner-cell' },
+        kind: GridCellKind.Custom,
+      };
+
+      if (row >= data.length) {
+        return customCell;
+      }
```
Comment (hkang1) [inline webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx:271]: question: should this just check to see if it's undefined as opposed to whether row is within the data length boundary? AKA:
```
if (!data[row]) return customCell;
```

**Review by hkang1 — APPROVED** [review]
LGTM, just one suggestion below

File: webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx (hunk 2 above)
Comment (keita-determined) [reply webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx:271]: i first wrote that way and change it to the current one. I just thought `row >= data.length` is more meaningful to code readers. I'll switch to `!data[row]` with a comment

Comment (ClaireNeveu) [reply webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx]: I agree that it's more meaningful to readers. Better to be explicit here.

Comment (keita-determined) [reply webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx]: added comment, so reader will know the context of the early return guard

**Pull request merged**